internal/provider/resource_tfe_variable.go

```diff
@@ -475,11 +534,28 @@ func (r *resourceTFEVariable) updateWithWorkspace(ctx context.Context, req resou
 		return
 	}
 	// Update state
-	result := modelFromTFEVariable(*variable, plan.Value)
+	result := modelFromTFEVariable(*variable, plan.Value, !config.ValueWO.IsNull())
+	updatePrivateState(ctx, resp, config.ValueWO)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
 	diags = resp.State.Set(ctx, &result)
 	resp.Diagnostics.Append(diags...)
 }
 
+func updatePrivateState(ctx context.Context, resp *resource.UpdateResponse, configValueWO types.String) {
```

Comment:
Consider making this a receiver method so that the `updatePrivateState` method name can be used in other resources.